gluonnlp/model/train/language_model.py

```diff
@@ -25,6 +25,157 @@
 from gluonnlp.model.utils import _get_rnn_layer
 from gluonnlp.model.utils import apply_weight_drop
 
+class MOSRNN(Block):
+    """Mos language model by Zhilin Yang, Zihang Dai, Ruslan Salakhutdinov, William W. Cohen.
+
+    Reference: https://github.com/zihangdai/mos
+
+    License: MIT License
+
+    Parameters
+    ----------
+    mode : str
+        The type of RNN to use. Options are 'lstm', 'gru', 'rnn_tanh', 'rnn_relu'.
+    vocab_size : int
+        Size of the input vocabulary.
+    embed_size : int
+        Dimension of embedding vectors.
+    hidden_size : int
+        Number of hidden units for RNN.
+    hidden_size_last : int
+        Number of hidden units for last RNN layer.
+    num_layers : int
+        Number of RNN layers.
+    tie_weights : bool, default False
+        Whether to tie the weight matrices of output dense layer and input embedding layer.
+    dropout : float
+        Dropout rate to use for encoder output.
+    weight_drop : float
+        Dropout rate to use on encoder h2h weights.
+    drop_h : float
+        Dropout rate to on the output of intermediate layers of encoder.
+    drop_i : float
+        Dropout rate to on the output of embedding.
+    drop_e : float
+        Dropout rate to use on the embedding layer.
+    drop_l : float
+        Dropout rate to use on the latent layer.
+    num_experts : int
+        Number of softmax.
+    """
+    def __init__(self, mode, vocab_size, embed_size=280, hidden_size=960, hidden_size_last=620,
+                 num_layers=3, tie_weights=False, dropout=0.2, weight_drop=0.5, drop_h=0.3,
+                 drop_i=0.55, drop_e=0.1, drop_l=0.3, num_experts=15, **kwargs):
+        super(MOSRNN, self).__init__(**kwargs)
+        self._mode = mode
+        self._vocab_size = vocab_size
+        self._embed_size = embed_size
+        self._hidden_size = hidden_size
+        self._hidden_size_last = hidden_size_last
+        self._num_layers = num_layers
+        self._dropout = dropout
+        self._drop_h = drop_h
+        self._drop_i = drop_i
+        self._drop_e = drop_e
+        self._drop_l = drop_l
+        self._num_experts = num_experts
+        self._weight_drop = weight_drop
+        self._tie_weights = tie_weights
+
+        with self.name_scope():
+            self.embedding = self._get_embedding()
+            self.encoder = self._get_encoder()
+            self.prior = nn.Dense(num_experts, use_bias=False, flatten=False)
+            self.latent = nn.Dense(num_experts * embed_size, 'tanh', flatten=False)
```

Comment:
It would be clearer if specifying the input size: `self.latent = nn.Dense(num_experts * embed_size, in_units=hidden_size_last, 'tanh', flatten=False)`